src/reportengine/api.py

```diff
@@ -0,0 +1,47 @@
+"""
+API for resource engine
+"""
+
+import logging
+import importlib
+
+from reportengine.resourcebuilder import ResourceBuilder
+from reportengine.resourcebuilder import FuzzyTarget
+from reportengine.configparser import Config
+from reportengine.environment import Environment
+
+log = logging.getLogger(__name__)
+
+class API:
+    """The API class"""
+    def __init__(self, providers, config_cls, env_cls, **kwargs):
+        prov_list = []
+        for prov in providers:
+            if isinstance(prov, str):
+                try:
+                    mod = importlib.import_module(prov)
+                except ImportError:
+                    log.error("Could not import module %s", prov)
+                    raise
+            else:
+                mod = prov
+            prov_list.append(mod)
+
+        self.provider_loaded = prov_list
+        self.config_class = config_cls
+        self.environment_class = env_cls
+        self.loadedenv = self.environment_class(**kwargs)
```

Comment:
At this point one is left wondering why don't we just pass an instance of environment in the first place.

Reply:
Probably would be fine, I had it like this so that `validphys.app.App` could inherit the environment class but probably this is unneccessary